okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -242,6 +242,11 @@ private void connectSocket(int connectTimeout, int readTimeout, Call call,
       throw ce;
     }
 
+    // Making sure rawSocket is not null due to rare hardware/firmware issues with creating/sustaining a socket.
+    if (rawSocket == null) {
```

Comment:
I think this is too late as its used already at this point (lines 236, 238)

Reply:
This condtition must be before line 235

Reply:
fixed